vita3k/util/src/util.cpp

```diff
@@ -267,6 +267,47 @@ void float_to_half(const float *src, std::uint16_t *dest, const int total) {
     }
 }
 
+#ifdef __AVX2__
+//forced use F16C instruction set
+//AVX2 checked intentionally cause checking AVX is not enough for some CPU architectures (Sandy bridge)
+void float_to_half(const float *src, std::uint16_t *dest, const int total) {
+    float_to_half_F16C(src, dest, total);
+}
+#else
+//check and use F16C instruction set if possible
+#include <util/instrset_detect.h>
+
+void float_to_half_basic(const float *src, std::uint16_t *dest, const int total) {
+    int i = 0;
+    while (i < total) {
+        uint32_t x = *((uint32_t *)&src[i]);
+        dest[i] = ((x >> 16) & 0x8000) | ((((x & 0x7f800000) - 0x38000000) >> 13) & 0x7c00) | ((x >> 13) & 0x03ff);
+        i++;
+    }
+}
+
+// use function variable as imitation of self-modifying code.
+// on first use we check processor features and set appropriate realisation, later we immediately use appropriate realisation
+void float_to_half_init(const float *src, std::uint16_t *dest, const int total);
+
+void (*float_to_half_var)(const float *src, std::uint16_t *dest, const int total) = float_to_half_init;
+
+void float_to_half_init(const float *src, std::uint16_t *dest, const int total) {
+    if (util::instrset::hasF16C()) {
```

Comment:
Requires AVX + F16C

Reply:
F16C include AVX (hasF16C function firstly check AVX)

Reply:
Miss that, all good then 👍